### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Accredited Representation Management, digital submission designs (Appoint a Representative v2, Find a Representative v3, Representative Status Widget v2)

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#95354]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Sorry for the delayed feedback. Please ping me in Slack if you have any questions!

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

### Find a Representative (sync)
- [x] **Must:** On the [pending/declined states for the widget](https://www.figma.com/design/bzbwObT9hiItve0q3cQX9c/Find-and-Appoint-a-Representative?node-id=9405-150014&node-type=frame&t=c78Iq3AfYNGojqeq-0), the placement of the alert before the heading means screen reader users who are navigating by heading are probably not going to hear that information. Typically, a screen reader user might jump to that heading and then have their screen reader start reading from that point onward --- and they wouldn't have any reason to believe there's relevant information that came before that point.
   
   Up to you on how best to address this. You could put the alert after the heading, or you could give the alert its own heading ("Your request is pending" or similar) so that it's more discoverable in its current location. I think Matt may have also have been hinting at a change to the widget itself to account for these states rather than placing an alert next to the widget. Any of these would be fine as far as I'm concerned.

### Appoint a Representative (async)
- [x] **Must:** Echoing @erinrwhite --- On the [search representatives screen](https://www.figma.com/design/bzbwObT9hiItve0q3cQX9c/Find-and-Appoint-a-Representative?node-id=6329-256216&t=1aMdxN4V8eL66kqJ-4), information about the results and status of the search should be displayed, ideally the same way that you're doing it for Find a Rep. Focus management should generally behave the same way as on Find a Rep as well.
- [x] **Must:** Each instance of the ["Select this representative"](https://www.figma.com/proto/bzbwObT9hiItve0q3cQX9c/Find-and-Appoint-a-Representative?page-id=5138%3A420728&node-id=8890-504749&t=JGlLMgAgrCeMcLKb-0&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=3717%3A80893&show-proto-sidebar=1) button must have a unique accessible name --- meaning that if a screen reader user tabs to that button, they should know which representative it's going to select without having to explore the surrounding context.
   
   The most direct way to do this is to change the button text to include the name of the person/org: "Select Arizona Department of Veterans Services" rather than "Select this representative." This would be my first choice.
   
   If that's not practical or desirable visually, you can also use an `aria-label` (the `va-button` component calls its prop `label`), so the visible text remains "Select this representative" but what's announced to screen reader users is unique. But nothing is ever easy! When a sighted voice command software speaks a command using the visible label ("Click select this representative"), the software often tries to match against the `aria-label` rather than the visible text. And if the start of the `aria-label` doesn't exactly match the visible text, some software won't give you a match.
   
   So an `aria-label` like "Select Arizona Department of Veterans Services as representative" would not be likely to get a match. An `aria-label` like "Select this representative, Arizona Department of Veterans Services" would be more likely to work, but it's definitely awkward. If you go the `aria-label` route, it might be worth tinkering with the visible label to see if there's a way to make the visible text and `aria-label` more grammatically compatible.
- [x] **Question/Must:** When you click ["Select this representative"](https://www.figma.com/proto/bzbwObT9hiItve0q3cQX9c/Find-and-Appoint-a-Representative?page-id=5138%3A420728&node-id=8890-504749&t=JGlLMgAgrCeMcLKb-0&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=3717%3A80893&show-proto-sidebar=1), does that immediately advance you to the next page in the flow (as happens in the prototype)? Or does it just select that card and you still have to click Continue?
   
   If it selects that card and you still have to click Continue, some thoughts:
   - Consider using a checkbox instead of a button. Not a requirement here, but it might make the interaction clearer.
   - Don't use color alone to indicate that a card has been selected.
   - If clicking the button again un-selects, make sure that's reflected in the button text.
- [x] **Must:** For the [error state on the search input](https://www.figma.com/proto/bzbwObT9hiItve0q3cQX9c/Find-and-Appoint-a-Representative?page-id=5138%3A420728&node-id=6772-281266&t=JGlLMgAgrCeMcLKb-0&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=3717%3A80893&show-proto-sidebar=1), the error message needs to be a distinct message, not just a re-statement of the instructions. Even if it's "You must enter a name" or something similar, the error message should give a direct indication of what the user needs to fix.

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [ ] Assign this ticket to the VFS team member that opened the Slack request
- [ ] Add the VFS team product label
- [ ] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [ ] Add the Collaboration Cycle initiative milestone

**Comment by oddball-lindsay** [comment]
> Question/Must: When you click ["Select this representative"](https://www.figma.com/proto/bzbwObT9hiItve0q3cQX9c/Find-and-Appoint-a-Representative?page-id=5138%3A420728&node-id=8890-504749&t=JGlLMgAgrCeMcLKb-0&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=3717%3A80893&show-proto-sidebar=1), does that immediately advance you to the next page in the flow (as happens in the prototype)? Or does it just select that card and you still have to click Continue?

@briandeconinck it will immediately advance, like the prototype indicates!
